docs/architecture/workflow-update.md

```diff
@@ -314,24 +317,35 @@ trying to recreate the Update.
 > #### TODO
 > This might be possible in the future when "Durable Admitted" Update is implemented.
 
-Accepted (but not Completed) Updates remain in this state after the Workflow is completed; since 
-they are already written to the event history. It is the Workflow's responsibility to complete them.
-This behavior is similar to Activities which can also still be completed after the Workflow was
-completed already (the result will be ignored).
+There two Workflow completion scenarios:
+- final completion, when no new run is created (e.g. with `COMPLETE_WORKFLOW_EXECUTION` command),
+- continued completion, when new run is created (e.g. `CONTINUE_AS_NEW_WORKFLOW_EXECUTION` command).
 
-When the Workflow completes, all in-flight Updates are aborted: admitted Updates get 
+When the Workflow performs a final completion, all in-flight Updates are aborted: admitted Updates get
 `ErrWorkflowCompleted` error on both `accepted` and `completed` futures, and accepted Updates get
-the same error on the `completed` future only.
+the same error on the `completed` future only. It is the Workflow's responsibility to complete
+accepted Updates. This behavior is similar to Activities which also can be left abandoned after 
+Workflow completed.
+
+If Workflow completion command creates a new run, accepted Updates are aborted in the same way:
+with `ErrWorkflowCompleted` error on the `completed` future. Admitted Updates, though, are aborted
+with retryable `ErrWorkflowClosing` error. SDK client retries this error and new attempt should land
+on the new run. This is because Updates that were received while the Workflow Task was running, haven't
+been seen by the Workflow yet, and can be safely retried on the new run. It also provides better
+experience for API callers: they shouldn't notice that the Workflow started a new run.
+
+Aborted accepted (but not completed) Updates remain in the `Aborted` state in the registry even after
+Update registry is reconstructed from the history.
```

Comment:
```suggestion
When a Workflow completion command creates a new run, accepted Updates are aborted in the same way:
with the `ErrWorkflowCompleted` error on the `completed` future. Admitted Updates, though, are aborted
with the retryable `ErrWorkflowClosing` error. The SDK client retries this error and a new attempt should land
on the new run. This is safe because Updates that were received while the Workflow Task was running
haven't been seen by the Workflow yet. It also provides a better experience
for API callers since they shouldn't notice that the Workflow started a new run.

Aborted accepted (but not completed) Updates remain in the `Aborted` state in the registry even after
the Update registry is reconstructed from the history.
```